lib/src/client/http2_channel.dart

```diff
@@ -31,27 +33,89 @@ class ClientChannel extends ClientChannelBase {
   final Object host;
   final int port;
   final ChannelOptions options;
+  final StreamController<ConnectionState> connectionStateStreamController =
```

Comment:
Can this live in `ClientChannelBase` instead? Then you don't need any of the duplicated code below. 

You would need to add a member to `ClientChannel` though to expose state changes in some form.

Reply:
Done in the new commit

Reply:
My suggestion was to move all of the code to the base class (this includes code from `terminate` and `shutdown` as well as attaching status change callback to the connection object).

Reply:
No rush.